Doppler.BigQueryMicroservice/Services/EmailSender.cs

```diff
@@ -0,0 +1,82 @@
+using Doppler.BigQueryMicroservice.Entitites.EmailSender;
+using Flurl.Http;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Tavis.UriTemplates;
+using Doppler.BigQueryMicroservice.Utils;
+using Microsoft.Extensions.Logging;
+
+namespace Doppler.BigQueryMicroservice.Services
+{
+    public class EmailSender : IEmailSender
+    {
+        private readonly ILogger<EmailSender> _logger;
+        private readonly RelayEmailSenderConfiguration _config;
+        private IFlurlClient _flurlClient = null;
+
+        private IFlurlClient FlurlClient => EnsureAndGetFlurlClient();
+
+        public EmailSender(IOptions<RelayEmailSenderConfiguration> config, ILogger<EmailSender> logger)
+        {
+            _config = config.Value;
+            _logger = logger;
+        }
+
+        ///
+        public async Task<bool> SendWithTemplateAsync(string templateId, object templateModel, IEnumerable<string> to, IEnumerable<string> cc = null,
+                                                IEnumerable<string> bcc = null, string fromName = null, string fromAddress = null, IEnumerable<Attachment> attachments = null,
+                                                CancellationToken cancellationToken = default)
```

Comment:
What do you think about using a simple indentation in place of a lot of spaces? it makes it easier to review in github.

![image](https://user-images.githubusercontent.com/1157864/134929021-4676245e-4987-4222-88e7-ed8a557bc5e3.png)

```suggestion
        public async Task<bool> SendWithTemplateAsync(
            string templateId,
            object templateModel,
            IEnumerable<string> to,
            IEnumerable<string> cc = null,
            IEnumerable<string> bcc = null,
            string fromName = null,
            string fromAddress = null,
            IEnumerable<Attachment> attachments = null,
            CancellationToken cancellationToken = default)
```

Reply:
suggest accepted and applied!!